internal/mode/static/nginx/modules/src/httpmatches.js

```diff
@@ -49,32 +48,37 @@ function redirect(r) {
   r.internalRedirect(match.redirectPath);
 }
 
-function extractMatchesFromRequest(r) {
-  if (!r.variables[MATCHES_VARIABLE]) {
-    throw Error(
-      `cannot redirect the request; the variable ${MATCHES_VARIABLE} is not defined on the request object`,
-    );
-  }
-
-  let matches;
-
+function internalRedirect(r, matchList) {
+  // matchList is a list of http matches in order of precedence.
```

Comment:
Looks like this comment belongs somewhere else

Reply:
moved it where we verify `matchList` now, hoepfully that makes more sense.